src/main/scala/GeomDataReader.scala

```diff
@@ -0,0 +1,164 @@
+package geotrellis.sdg
+
+import geotrellis.proj4._
+import geotrellis.proj4.util._
+import geotrellis.vector.{Extent, Geometry}
+import geotrellis.vector.reproject._
+import geotrellis.vectortile._
+import geotrellis.layer._
+
+import org.locationtech.geomesa.spark.jts._
+
+import org.locationtech.jts.geom.{Geometry => JTSGeometry}
+
+import org.apache.commons.io.IOUtils
+
+import org.apache.spark.sql.{SQLContext, DataFrame}
+import org.apache.spark.sql.functions.{udf, explode, lit}
+
+import java.io.{ByteArrayInputStream, ByteArrayOutputStream}
+import java.net.URI
+import java.util.zip.GZIPInputStream
+
+
+object GeomDataReader {
+  def readAndFormat(
+    sqlContext: SQLContext,
+    targetPath: String,
+    countryCode: String,
+    transform: MapKeyTransform
```

Comment:
As we discussed, this is actually fixed for MBTiles or at very least depends on the `zoom` column (if it ever changes). It's not appropriate to make this a parameter because its not and you (the function author) know the exact right value for it.